### ISSUE TITLE
Regenerating of New SSL Certificates

### ISSUE THREAD

**Issue by BPJOHN1990** [wazuh/wazuh-packages#2673]
Hi all,

I have followed the instructions here and created the certificates accordingly. https://documentation.wazuh.com/current/user-manual/certificates.html. However I need to change the SSL certificates on port 1515 and 55000. I tried looking for documentation on it and could not find any, appreciate if anyone could point me to the right direction.

**Comment by davidcr01** [comment]
Hello. I guess you are trying to modify the SSL certificates of the Wazuh manager. To perform this, we have some related documentation that can help: https://documentation.wazuh.com/current/user-manual/agent-enrollment/security-options/manager-identity-verification.html. This documentation explains how to generate and customize the SSL certificates of the Wazuh manager, among others.

Hope it helps.

**Comment by BPJOHN1990** [comment]
Hi @davidcr01,

Appreciate the link, but can I get more details for provided URL? Because both of my SSL certs are expired on Port 1515 and 55000. The steps is unclear on which file to generate and replace? Also where is req.conf?

Would appreciate your kind help if you can let me know how to replace the SSL certificates on Port 1515 and 55000

**Comment by davidcr01** [comment]
Hello again.

Firstly, the 1515 and the 55000 ports are related to the Wazuh manager and the Wazuh agent enrollment. You can get more information in [this link](https://documentation.wazuh.com/current/getting-started/architecture.html#required-ports). So then, if you want to renew the certificates of the 1515 and the 55000 port, then it means that you want to renew the certificates of the Wazuh manager. This is related to the documentation I have attached you in the previous reply.

The `req.conf` file, in the context of OpenSSL, is a configuration file used to provide default values and settings when generating certificate signing requests (CSRs) or certificates. It is commonly used with the openssl req command.

When you generate a CSR using the `openssl` req command, OpenSSL will look for configuration settings in the `req.conf` file unless explicitly specified in the command line. The `req.con`f file allows you to set default values for various parameters, such as country, state, organization, and common name. So, to renew the Wazuh manager certificates, you should create this file.

In this case, you should replace the `MANAGER_IP` variable with your Wazuh manager IP address in the `req.conf` file, as specified in the documentation.

Then, you should create a certificate signing request and sign that request with the following commands:
```console
openssl req -new -nodes -newkey rsa:4096 -keyout sslmanager.key -out sslmanager.csr -config req.conf

openssl x509 -req -days 365 -in sslmanager.csr -CA rootCA.pem -CAkey rootCA.key -out sslmanager.cert -CAcreateserial -extfile req.conf -extensions req_ext
```
Take into account that the RootCA key and certificate is used in the previous command. These files should be located in the same machine where you generated the Wazuh certificates. Then, you must copy the new Wazuh manager certificate to the right folder and restart the service as follows:

```console
cp sslmanager.cert sslmanager.key /var/ossec/etc
systemctl restart wazuh-manager
```

**Comment by BPJOHN1990** [comment]
Hi David,

Much appreciated, Port 1515 is resolved following your steps however, for port 55000 was still not fixed yet. Any idea on how to fix for port 55000?

**Comment by davidcr01** [comment]
Hello. By default, the Wazuh manager uses the 55000 port, and uses the following configuration: https://documentation.wazuh.com/current/user-manual/api/configuration.html#https. Related to this, it uses TLSv1.2.

Related to this, we have the following issue closed that reported that a warning was being generated reporting the deprecation of the TLSv.2 case: https://github.com/wazuh/wazuh/issues/20396
This is fixed in the following PR, which was merged in the incoming 4.8.0 new Wazuh version: https://github.com/wazuh/wazuh/pull/20420
